autorest/azure/client.go

```diff
@@ -0,0 +1,62 @@
+package azure
+
+import (
+	"net/http"
+	"net/http/cookiejar"
+
+	"github.com/Azure/go-autorest/autorest"
+)
+
+var (
+	statusCodesForRetry = append(autorest.StatusCodesForRetry, http.StatusConflict)
+)
+
+// Client is the Azure specific implementation for autorest.Sender
+type Client struct {
+	autorest.Client
+}
+
+// Do implements the Sender interface by invoking the active Sender after applying authorization.
+// If Sender is not set, it uses a new instance of http.Client. In both cases it will, if UserAgent
+// is set, apply set the User-Agent header. This is the Azure specific implementation
+func (azc Client) Do(r *http.Request) (*http.Response, error) {
+	r, err := autorest.Prepare(r,
+		azc.AzurePreparation(),
+	)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := autorest.SendWithSender(azc.sender(), r,
+		DoRetryForStatusCodes(azc, statusCodesForRetry...))
+	autorest.Respond(resp,
+		azc.ByInspecting())
+	return resp, err
+}
+
+// sender returns the Sender to which to send requests.
+func (azc Client) sender() autorest.Sender {
+	if azc.Sender == nil {
+		j, _ := cookiejar.New(nil)
+		return &http.Client{Jar: j}
+	}
+	return azc.Sender
+}
+
+// AzurePreparation is a prepare decorator that adds a User Agent and Authorization headers to the request
+func (azc Client) AzurePreparation() autorest.PrepareDecorator {
```

Comment:
I don't think this needs to be exported does it?

Reply:
Fixed